File: cinema/src/main/java/com/example/cinema/availableSeats/AvailableSeats.java
package com.example.cinema.availableSeats;

import com.example.cinema.seance.Seance;
import com.example.cinema.seat.Seat;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import static jakarta.persistence.FetchType.EAGER;
import static jakarta.persistence.FetchType.LAZY;

@Entity
@Setter
@Getter
public class AvailableSeats {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne(fetch = EAGER)
    private Seance seance;
    @ManyToOne(fetch = EAGER)
    private Seat seat;
    @Enumerated(EnumType.STRING)
    private SeatStatus seatStatus;
    private Double price;
}

File: cinema/angularclient/src/app/model/available-seats.ts
import { MovieSeance } from "./movie-seance";
import { Seat } from "./seat";

export class AvailableSeats {
    id: number;
    seance: MovieSeance;
    seat: Seat;
    seatStatus: string;
    price: number;
    selected: boolean;

    constructor() {
        this.id = 0;
        this.seat = new Seat();
        this.seance = new MovieSeance();
        this.seatStatus = "";
        this.price= 0;
        this.selected = false;
    }
}
